casadocodigo/src/main/java/br/com/deveficiente/casadocodigo/livro/LivroController.java

```diff
@@ -50,5 +59,37 @@ public LivroDetalheDTO detalharLivroPorId(@PathVariable("id") Long id) {
 		return new LivroDetalheDTO(livro);
 	}
 
+	@GetMapping(value = "/api/carrinho/{id}")
+	public String adicionarLivroCarrinho(@PathVariable("id") Long id, @CookieValue("carrinho") Optional<String> carrinho, HttpServletResponse response) {
+		try {
+			Carrinho novoCarrinho;
+			if(carrinho.isPresent()) {
+				novoCarrinho = new ObjectMapper().readValue(carrinho.get(), Carrinho.class);
+			} else {
+				novoCarrinho = new Carrinho();
+			}
+			novoCarrinho.adicionarLivroNoCarrinho(livroRepository.findById(id).get());
+			Cookie cookie = new Cookie("carrinho", new ObjectMapper().writeValueAsString(novoCarrinho));
+			response.addCookie(cookie);
+			
+			return novoCarrinho.toString();
+		} catch (JsonProcessingException e) {
+			throw new RuntimeException(e);
+		}
+		
+	}
+	
+	//TODO Ainda falta fazer funcionar, no momento não encontra nenhum cookie :(
+	@GetMapping(value = "/api/cookie/{id}")
+	public String consultarCookie(@PathVariable("id") Long id ,HttpServletRequest request) {
```

Comment:
*Deixamos pistas que facilitem o uso do código onde não conseguimos resolver com compilação.*

Aqui você pode deixar um nome de método e endpoint mais indicativo.

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Pode usar a mesma annotation @CookieValue para receber o valor do cookie aqui.

Reply:
Sobre não achar o cookie, confere primeiro o básico. O valor do cookie está sendo escrito na resposta? Na requisição o valor escrito do cookie ta indo? Vai verificando de fora para dentro até isolar o problema.